src/components/course/EnrollModal.jsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Modal } from '@edx/paragon';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faSpinner } from '@fortawesome/free-solid-svg-icons';
+
+export const modalText = {
+  noOffers: {
+    body: 'You do not have any applicable vouchers, but you can still enroll in this course.',
+    button: 'Continue to payment',
+    title: 'Payment required for course enrollment',
+  },
+  fullOffers: {
+    body: (offersCount) => `Enrolling in this course will use 1 of your ${offersCount} remaining vouchers.`,
+    button: 'Enroll in course',
+    title: 'Use 1 voucher for this course',
+  },
+};
+
+const EnrollModal = ({
+  courseHasOffer,
+  enrollmentUrl,
+  isModalOpen,
+  offersCount,
+  setIsModalOpen,
```

Comment:
not major but maybe keep isModalOpen, and setIsModalOpen next to each other here?

Reply:
I, uh, got an alphebetizing add-on for vscode and may have fallen in love with it a little too hard.